xarray/core/alignment.py

```diff
@@ -31,10 +31,34 @@ def _get_joiner(join):
         # We cannot return a function to "align" in this case, because it needs
         # access to the dimension name to give a good error message.
         return None
+    elif join == "override":
+        # We rewrite all indexes and then use join='left'
+        return operator.itemgetter(0)
     else:
         raise ValueError("invalid value for join: %s" % join)
 
 
+def _override_indexes(objects, all_indexes, exclude):
+    for dim, dim_indexes in all_indexes.items():
+        if dim not in exclude:
+            lengths = [dd.size for dd in dim_indexes]
```

Comment:
Could we just use a set comprehension here? Also I have a minor distaste for abbreviations like `dd`.

e.g., maybe
``python
            lengths = {index.size for index in dim_indexes}
            if len(lengths) != 1:
                raise ValueError(...)
```

Reply:
I like that.